src/write.c

```diff
@@ -1092,52 +1111,17 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         }
 
         if (encoder->data->alphaPresent) {
-            uint16_t gridAlphaID = 0;
-            if (cellCount > 1) {
-                avifEncoderItem * gridAlphaItem = avifEncoderDataCreateItem(encoder->data, "grid", "Alpha", 6, 0);
-                avifWriteGridPayload(&gridAlphaItem->metadataPayload, gridCols, gridRows, gridWidth, gridHeight);
-                gridAlphaItem->alpha = AVIF_TRUE;
-                gridAlphaItem->irefToID = encoder->data->primaryItemID;
-                gridAlphaItem->irefType = "auxl";
-                gridAlphaItem->gridCols = gridCols;
-                gridAlphaItem->gridRows = gridRows;
-                gridAlphaItem->gridWidth = gridWidth;
-                gridAlphaItem->gridHeight = gridHeight;
-                gridAlphaID = gridAlphaItem->id;
-
-                if (encoder->data->imageMetadata->alphaPremultiplied) {
-                    avifEncoderItem * primaryItem = avifEncoderDataFindItemByID(encoder->data, encoder->data->primaryItemID);
-                    assert(primaryItem);
-                    primaryItem->irefType = "prem";
-                    primaryItem->irefToID = gridAlphaID;
-                }
-            }
-
-            for (uint32_t cellIndex = 0; cellIndex < cellCount; ++cellIndex) {
-                avifEncoderItem * item = avifEncoderDataCreateItem(encoder->data, "av01", "Alpha", 6, cellIndex);
-                item->codec = avifCodecCreate(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
-                if (!item->codec) {
-                    return AVIF_RESULT_NO_CODEC_AVAILABLE;
-                }
-                item->codec->csOptions = encoder->csOptions;
-                item->codec->diag = &encoder->diag;
-                item->alpha = AVIF_TRUE;
-                item->extraLayerCount = encoder->extraLayerCount;
-
-                if (cellCount > 1) {
-                    item->dimgFromID = gridAlphaID;
-                    item->hiddenImage = AVIF_TRUE;
-                } else {
-                    item->irefToID = encoder->data->primaryItemID;
-                    item->irefType = "auxl";
-
-                    if (encoder->data->imageMetadata->alphaPremultiplied) {
-                        avifEncoderItem * primaryItem = avifEncoderDataFindItemByID(encoder->data, encoder->data->primaryItemID);
-                        assert(primaryItem);
-                        primaryItem->irefType = "prem";
-                        primaryItem->irefToID = item->id;
-                    }
-                }
+            uint16_t alphaItemID;
+            AVIF_CHECKRES(avifEncoderAddImageItems(encoder, gridCols, gridRows, gridWidth, gridHeight, /*alpha=*/AVIF_TRUE, &alphaItemID));
+            avifEncoderItem * alphaItem = avifEncoderDataFindItemByID(encoder->data, alphaItemID);
```

Comment:
Just a note: I think it is possible to avoid this extra `avifEncoderDataFindItemByID()` call by moving lines 1118-1119 into the `avifEncoderAddImageItems()` function. But it will require a fixed order of the two `avifEncoderAddImageItems()` calls -- first color, then alpha -- so that `encoder->data->primaryItemID` has the correct value during the call for alpha.